beacon-chain/blockchain/van_process_pending_blocks.go

```diff
@@ -220,16 +225,43 @@ func (s *Service) waitForConfirmation(b interfaces.SignedBeaconBlock) error {
 	}
 }
 
-// verifyPandoraShardInfo
+// verifyPandoraShardInfo verifies Pandora shard info
 func (s *Service) verifyPandoraShardInfo(parentBlk, curBlk interfaces.SignedBeaconBlock) error {
+	var parentPanShards []*ethpb.PandoraShard
+
+	// Current beacon block
+	if curBlk == nil || curBlk.IsNil() {
+		return errUnknownCurrent
+	}
+	curBlkObj := curBlk.Block()
+	if curBlkObj.IsNil() {
+		return errUnknownCurrent
+	}
 	// For slot #1, we don't have shard info for previous block so short circuit here
-	if curBlk.Block().Slot() == 1 {
+	if curBlkObj.Slot() == 1 {
 		return nil
 	}
-	// Checking length of current block's pandora shard info
-	curPanShards := curBlk.Block().Body().PandoraShards()
-	parentPanShards := parentBlk.Block().Body().PandoraShards()
+	curBlkObjBody := curBlkObj.Body()
```

Comment:
We do not need to check body because if you see `IsNil()` method, you can see this method already checks this

Reply:
we need to otherwise you break the law of demeter

Reply:
@atif-konasl There are only option that somehow someone will give me `nil` interface. So because `IsNil()` is checking only `w.b == nil || w.Body().IsNil()` not the interface `w` then I've just secured this too.

Reply:
I'm leaving this guard:
```go
	// Current beacon block
	if curBlk == nil || curBlk.IsNil() {
		return errUnknownCurrent
	}
```

Reply:
@atif-konasl So as we discussed, because of `prysmaticlabs` implementation I should remove 
```
curBlkObjBody := curBlkObj.Body()
if curBlkObjBody.IsNil() {
	return errUnknownCurrentBody
}
````
 because ` curBlk.IsNil()` checks for  `curBlk.Block().IsNil()` and `curBlk.Block().Body().isNil()` and just provide `curPanShards := curBlk.Block().Body().PandoraShards()`, right?

Reply:
@atif-konasl I solved this by providing above implementation changes. You can re-review now.